packages/salesforcedx-vscode-core/test/jest/commands/deployExecutor.test.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode';
+import { ComponentSet } from '@salesforce/source-deploy-retrieve';
+import * as fs from 'fs';
+import { DeployExecutor } from '../../../src/commands/DeployExecutor';
+import { WorkspaceContext } from '../../../src/context/workspaceContext';
+import { SourceTrackingService } from '../../../src/services';
+
+const dummyProjectPath = '/a/project/path';
+jest.mock('@salesforce/source-deploy-retrieve', () => {
+  return {
+    ...jest.requireActual('@salesforce/source-deploy-retrieve'),
+    getRootWorkspacePath: () => dummyProjectPath,
+    ComponentSet: jest.fn().mockImplementation(() => {
+      return {
+        deploy: jest.fn().mockImplementation(() => {
+          return { pollStatus: jest.fn() };
+        }),
+        getSourceComponents: jest.fn().mockReturnValue([
+          { name: '1', type: 'ApexClass' },
+          { name: '2', type: 'ApexClass' }
+        ])
+      };
+    })
+  };
+});
+
+jest.mock('@salesforce/salesforcedx-utils-vscode', () => {
+  return {
+    ...jest.requireActual('@salesforce/salesforcedx-utils-vscode'),
+    getRootWorkspacePath: () => dummyProjectPath,
+    ChannelService: jest.fn().mockImplementation(() => {
+      return {};
+    }),
+    TelemetryService: { getInstance: jest.fn() },
+    TelemetryBuilder: jest.fn()
+  };
+});
+
+jest.mock('../../../src/messages', () => {
+  return { loadMessageBundle: jest.fn(), nls: { localize: jest.fn() } };
+});
+
+describe('Deploy Executor', () => {
+  const dummyProcessCwd = '/';
+  const workspaceContextGetInstanceSpy = jest.spyOn(
+    WorkspaceContext,
+    'getInstance'
+  );
+  const mockWorkspaceContext = { getConnection: jest.fn() } as any;
+  const createSourceTrackingSpy = jest.spyOn(
+    SourceTrackingService,
+    'createSourceTracking'
+  );
+  const dummyComponentSet = new ComponentSet();
+  const deploySpy = jest.spyOn(dummyComponentSet, 'deploy');
```

Comment:
These mocks seems like they should go in the beforeEach. That way when another test is added they are reset on every run.

Reply:
either that or the top of the it if only applicable to that test